src/spclient.rs

```diff
@@ -291,17 +291,12 @@ impl SpClient {
         scan_sk: SecretKey,
         spend_key: SpendKey,
         mnemonic: Option<String>,
-        is_testnet: bool,
+        network: Network,
```

Comment:
I think it's better if we use bitcoin::Network for this. The 'Network' type from the silentpayments crate refers to the address type rather than the network used, which is why there is no 'signet' option. But in SpClient, I think the client should work with the actual network.

Reply:
Updated, That's a bit ugly though, and I think at some point we'd better have some conversion functions directly implemented in `silentpayments::utils::Network`. Maybe we could use the `&str` core arguments as some kind of pivot to avoid having to add bitcoin dependency just for that